cumulusci/tasks/salesforce/users/tests/test_photos.py

```diff
@@ -0,0 +1,372 @@
+        ]
+
+        # Execute the test
+        with pytest.raises(CumulusCIException) as e:
+            self.task._get_user_id_by_query(where)
+
+        assert (
+            e.value.args[0] == f"More than one User found (at least 2): {id_0}, {id_1}"
+        )
+
+        self.task.sf.query_all.assert_called_once_with(expected_query)
+        self.task.logger.info.assert_has_calls(self.task.logger.info._expected_calls)
+
+    def test_get_user_id_by_query__invalid_query(self):
+        where = "(FirstName = 'User' OR LastName = 'User') AND IsActive = true AND CreatedDate = TODAY but the end of this query is invalid SOQL "
+        expected_query = "SELECT Id FROM User WHERE (FirstName = 'User' OR LastName = 'User') AND IsActive = true AND CreatedDate = TODAY but the end of this query is invalid SOQL  LIMIT 2"
+
+        self.task.sf.query_all.side_effect = self.e
+
+        self.task.logger.info._expected_calls = [
+            mock.call(f"Querying User: {expected_query}")
+        ]
+
+        # Execute the test
+        with pytest.raises(CumulusCIException) as e:
+            self.task._get_user_id_by_query(where)
+
+        assert e.value.args[0] == self.expected_exception_message
+
+        self.task.sf.query_all.assert_called_once_with(expected_query)
+        self.task.logger.info.assert_has_calls(self.task.logger.info._expected_calls)
+
+    def test_get_default_user_id(self):
+        expected = self.user_id
+
+        self.task.sf.restful.return_value = {
+            "identity": f"a bunch of other things{expected}"
+        }
+
+        actual = self.task._get_default_user_id()
+
+        assert expected == actual
+
+        self.task.logger.info.assert_called_once_with(
+            f"Uploading profile photo for the default User with ID {expected}"
+        )
+
+    def test_insert_content_document__no_photo_found(self):
+        with temporary_dir():
+            with pytest.raises(CumulusCIException) as e:
+                self.task._insert_content_document(self.photo)
+
+            assert e.value.args[0] == f"No photo found at {self.photo}"
+
+    def test_insert_content_document__exception_creating_content_document(self):
+        errors = "the reason why inserting the ContentVersion failed"
+        self.task.sf.ContentVersion.create.return_value = {
+            "success": False,
+            "errors": errors,
+        }
+
+        with temporary_dir() as d:
+            # Copy photo to temporary direcory.
+            shutil.copy(str(self.photo_path), d)
+            temp_photo_path = pathlib.Path(os.path.join(d, self.photo)).resolve()
+            assert (
+                temp_photo_path.exists()
+            ), "photo mock was not copied to the temporary directory"
+            assert (
+                temp_photo_path.is_file()
+            ), "Path the the photo mock in the temporary directory should point to a file"
+
+            self.task.sf.ContentVersion.create._expected_calls = [
```

Comment:
The intention is to attach the expected called right to the mock to keep everything in one variable.  I've found it useful when there are several mocked methods.  But since this creates confusion, I won't reuse this technique and probably create separate local variables.